app/controllers/orders_controller.rb

```diff
@@ -7,4 +7,18 @@ def show
     @order = Order.find(params[:id])
   end
 
+  def create
+    user = User.find(session[:user_id].to_i)
+
+    order = user.orders.create(total: @cart.total_price.to_s)
+
+    @cart.contents.each do |id, quantity|
+      order.order_creatures.create(creature: Creature.find(id),
```

Comment:
I believe you could do something like this:

in your cart class you could have a method that will return all the creature_ids 

```
def creature_ids
     contents.keys
end
```

Then in the controller: 
```
user.orders.create(total: @cart.total_price, creatures: Creature.find(@cart.creature_ids))
```